pkg/manifest/a11y.go

```diff
@@ -0,0 +1,407 @@
+	A11yFeatureHighContrastDisplay A11yFeature = "highContrastDisplay"
+
+	// The content has been formatted to meet large print guidelines.
+	//
+	// The property is not set if the font size can be increased. See DisplayTransformability.
+	A11yFeatureLargePrint A11yFeature = "largePrint"
+
+	// The content is in braille format, or alternatives are available in braille.
+	A11yFeatureBraille A11yFeature = "braille"
+
+	// When used with creative works such as books, indicates that the resource includes
+	// tactile graphics. When used to describe an image resource or physical object,
+	// indicates that the resource is a tactile graphic.
+	A11yFeatureTactileGraphic A11yFeature = "tactileGraphic"
+
+	// When used with creative works such as books, indicates that the resource includes models
+	// to generate tactile 3D objects. When used to describe a physical object,
+	// indicates that the resource is a tactile 3D object.
+	A11yFeatureTactileObject A11yFeature = "tactileObject"
+
+	// Indicates that the resource does not contain any accessibility features.
+	A11yFeatureNone A11yFeature = "none"
+)
+
+func A11yFeaturesFromStrings(strings []string) []A11yFeature {
+	return fromStrings(strings, func(str string) A11yFeature {
+		return A11yFeature(str)
+	})
+}
+
+// A11yHazard is a characteristic of the described resource that is physiologically dangerous to some users.
+type A11yHazard string
+
+const (
+	// Indicates that the resource presents a flashing hazard for photosensitive persons.
+	A11yHazardFlashing A11yHazard = "flashing"
+
+	// Indicates that the resource does not present a flashing hazard.
+	A11yHazardNoFlashingHazard A11yHazard = "noFlashingHazard"
+
+	// Indicates that the resource contains instances of motion simulation that
+	// may affect some individuals.
+	//
+	// Some examples of motion simulation include video games with a first-person perspective
+	// and CSS-controlled backgrounds that move when a user scrolls a page.
+	A11yHazardMotionSimulation A11yHazard = "motionSimulation"
+
+	// Indicates that the resource does not contain instances of motion simulation.
+	A11yHazardNoMotionSimulationHazard A11yHazard = "noMotionSimulationHazard"
+
+	// Indicates that the resource contains auditory sounds that may affect some individuals.
+	A11yHazardSound A11yHazard = "sound"
+
+	// Indicates that the resource does not contain auditory hazards.
+	A11yHazardNoSoundHazard A11yHazard = "noSoundHazard"
+
+	// Indicates that the author is not able to determine if the resource presents any hazards.
+	A11yHazardUnknown A11yHazard = "unknown"
+
+	// Indicates that the resource does not contain any hazards.
+	A11yHazardNone A11yHazard = "none"
+)
+
+func A11yHazardsFromStrings(strings []string) []A11yHazard {
+	return fromStrings(strings, func(str string) A11yHazard {
+		return A11yHazard(str)
+	})
+}
+
+func fromStrings[T any](strings []string, transform func(string) T) []T {
+	res := []T{}
```

Comment:
There is an optimization that can be made here to reduce allocations from growing the slice:
```go
res := make([]T, 0, len(strings))
```

Reply:
Makes sense, thanks.